### PR TITLE
Create more structures and fixed channels

### PR DESCRIPTION
**Pull request by Cqdet**
I think the channels can definitely be improved upon. The initial plan was to have `Channel` just be the extensible channel to the rest of the other channels (i.e `TextChannel`, `VoiceChannel`, `PrivateChannel`, etc.), but I ran into a bug that ended up just not helping and caused me simply to make `GuildChannel`.

Anyways, I think the only thing that really needs to be changed is the Event Handler since I never took into account the caching of guilds and channels so values may return undefined.

### CODE REVIEW

File: src/structures/channel/TextChannel.ts [hunk 1]
```diff
@@ -3,51 +3,48 @@ import MessageContent from "../../network/discord/interfaces/MessageContent.ts";
...
-     public position: number;
-     public permissionOverwrites: any[];
-     public rateLimit: number;
-     public nsfw: boolean;
+class TextChannel extends GuildChannel {
+     public guild: Guild | undefined;
+     public rateLimitPerUser: number;
+     public lastPinTimestamp: number
      public topic: string;
-     public lastMessageId: string;
```
Comment (Bavfalcon9) [inline src/structures/channel/TextChannel.ts:15]: why?

File: src/structures/channel/TextChannel.ts [hunk 2]
```diff
@@ -3,51 +3,48 @@ import MessageContent from "../../network/discord/interfaces/MessageContent.ts";
...
-          this.permissionOverwrites = data.permission_overwrites;
-          this.rateLimit = data.rate_limit_per_user;;
-          this.nsfw = data.nsfw || false;
+     constructor(data: any, guild?: Guild) {
+          super(data);
+          this.guild = guild
+          this.rateLimitPerUser = data.rate_limit_per_user;;
           this.topic = data.topic;
-          this.lastMessageId = data.last_message_id;
-          this.parentId = data.parent_id;
```
Comment (Bavfalcon9) [inline src/structures/channel/TextChannel.ts:28]: This exists on text channels.

File: src/structures/guild/Guild.ts [hunk 3]
```diff
@@ -70,16 +77,36 @@ class Guild extends Base {
...
         this.preferredLocale = data.preferred_locale;
         this.premiumTier = data.premium_tier;
-        this.subscriptionCount = data.premium_subscription_count || 0;
-        // annnnnddddd i dont wanna do this right now :(
+        this.subscriptionCount = data.premium_subscription_count || 0;   
     }
 
-    public get paritial(): boolean {
-        return false;
+    public static dummyObject(): Guild {
```
Comment (Bavfalcon9) [inline src/structures/guild/Guild.ts:87]: These are supposed to be temporary.

File: src/structures/guild/Guild.ts [hunk 4]
```diff
@@ -70,16 +77,36 @@ class Guild extends Base {
...
+             afkTimeout: null,
+        });
+   }
+
+    public get paritial(): boolean { return false }
+
+    /**
+     * Guild Action Methods
+     */
+    public async ban(mid: string, deleteMessagesDays?: 1 | 2 | 3 | 4 | 5 | 6 | 7, reason?: string): Promise<void> {
```
Comment (Bavfalcon9) [inline src/structures/guild/Guild.ts:106]: Should return a guildBan instance.

File: src/structures/guild/Guild.ts [hunk 5]
```diff
@@ -70,16 +77,36 @@ class Guild extends Base {
...
+    public get paritial(): boolean { return false }
+
+    /**
+     * Guild Action Methods
+     */
+    public async ban(mid: string, deleteMessagesDays?: 1 | 2 | 3 | 4 | 5 | 6 | 7, reason?: string): Promise<void> { 
+        await RequestManager.banMember(this.id, mid, deleteMessagesDays, reason)
     }
+
+    public async kick(mid: string, reason?: string): Promise<void> { await RequestManager.kickMember(this.id, mid, reason) }
```
Comment (Bavfalcon9) [inline src/structures/guild/Guild.ts:110]: await not needed

File: src/structures/guild/Guild.ts [hunk 6]
```diff
@@ -70,16 +77,36 @@ class Guild extends Base {
...
+        });
+   }
+
+    public get paritial(): boolean { return false }
+
+    /**
+     * Guild Action Methods
+     */
+    public async ban(mid: string, deleteMessagesDays?: 1 | 2 | 3 | 4 | 5 | 6 | 7, reason?: string): Promise<void> { 
+        await RequestManager.banMember(this.id, mid, deleteMessagesDays, reason)
```
Comment (Bavfalcon9) [inline src/structures/guild/Guild.ts:107]: await not needed

**Review by Bavfalcon9 — CHANGES_REQUESTED** [review]

**Review by Bavfalcon9 — APPROVED** [review]
Some of this will be changed. also next time keep the SAME syntax. especially with imports....

**Comment by Cqdet** [comment]
Are the caching issues fixed as well?

**Pull request merged**